tools/crates/build_crate.gni

```diff
@@ -3,60 +3,85 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import("//brave/build/host_os_constants.gni")
+
 template("build_crate") {
-  crate = target_name
-  assert(
-      defined(invoker["${crate}_exe"]),
-      "Please add ${crate}_exe to //brave/tools/crates/config.gni (or to $crate itself)!")
-  crate_exe = invoker["${crate}_exe"]
-  crate_target_dir = get_path_info(get_path_info(crate_exe, "dir"), "dir")
-
-  action("build_$crate") {
+  if (defined(invoker["crates"])) {
+    crates = invoker["crates"]
+  } else {
+    crates = [ target_name ]
+  }
+
+  if (defined(invoker["executable_outputs"])) {
+    executable_outputs = invoker["executable_outputs"]
+  } else {
+    executable_outputs = [ target_name ]
+  }
+
+  cargo_target_dir = "$root_build_dir/brave_tools_crates"
```

Comment:
This could turn into a race. All `build_crate` targets would share the same `cargo_target_dir = "$root_build_dir/brave_tools_crates"`, but declare different executable outputs. GN would see no conflict and might run them in parallel, even though they are all poking at the same Cargo target directory.